pyaedt/edb_grpc/core/edb_data/components_data.py

```diff
@@ -0,0 +1,962 @@
+
+
+class EDBComponentDef(object):
+    """Manages EDB functionalities for component definitions.
+
+    Parameters
+    ----------
+    parent : :class:`pyaedt.edb_grpc.core.components.Components`
+        Inherited AEDT object.
+    comp_def : object
+        Edb ComponentDef Object
+    """
+
+    def __init__(self, components, comp_def):
+        self._pcomponents = components
+        self._edb_comp_def = comp_def
+
+    @property
+    def _comp_model(self):
+        return list(self._edb_comp_def.GetComponentModels())  # pragma: no cover
+
+    @property
+    def part_name(self):
+        """Retrieve component definition name."""
+        return self._edb_comp_def.GetName()
+
+    @part_name.setter
+    def part_name(self, name):
+        self._edb_comp_def.SetName(name)
+
+    @property
+    def type(self):
+        """Retrieve the component definition type.
+
+        Returns
+        -------
+        str
+        """
+        num = len(set(comp.type for refdes, comp in self.components.items()))
+        if num == 0:  # pragma: no cover
+            return None
+        elif num == 1:
+            return list(self.components.values())[0].type
+        else:
+            return "mixed"  # pragma: no cover
+
+    @type.setter
+    def type(self, value):
+        for comp in list(self.components.values()):
+            comp.type = value
+
+    @property
+    def components(self):
+        """Get the list of components belonging to this component definition.
+
+        Returns
+        -------
+        dict of :class:`pyaedt.edb_grpc.core.edb_data.components_data.EDBComponent`
+        """
+        comp_list = [
+            EDBComponent(self._pcomponents, l)
+            for l in self._pcomponents._edb.Cell.Hierarchy.Component.FindByComponentDef(
+                self._pcomponents._pedb.active_layout, self.part_name
+            )
+        ]
+        return {comp.refdes: comp for comp in comp_list}
+
+    @pyaedt_function_handler
+    def assign_rlc_model(self, res, ind, cap, is_parallel=False):
+        """Assign RLC to all components under this part name.
+
+        Parameters
+        ----------
+        res : int, float
+            Resistance.
+        ind : int, float
+            Inductance.
+        cap : int, float
+            Capacitance.
+        is_parallel : bool, optional
+            Whether it is parallel or series RLC component.
+        """
+        for comp in list(self.components.values()):
+            res, ind, cap = res, ind, cap
+            comp.assign_rlc_model(res, ind, cap, is_parallel)
+        return True
+
+    @pyaedt_function_handler
+    def assign_s_param_model(self, file_path, model_name=None, reference_net=None):
+        """Assign S-parameter to all components under this part name.
+
+        Parameters
+        ----------
+        file_path : str
+            File path of the S-parameter model.
+        name : str, optional
+            Name of the S-parameter model.
```

Comment:
```suggestion
            Name of the S-parameter model.
            
```
Blank line needed before Returns section. A description of what is returned is still needed.